habitat/utils/geometry_utils.py

```diff
@@ -51,6 +53,81 @@ def quaternion_wxyz_to_xyzw(v: np.array):
 
 
 def quaternion_to_list(q: np.quaternion):
-    return quaternion.as_float_array(
-        quaternion_wxyz_to_xyzw(quaternion.as_float_array(q))
-    ).tolist()
+    r"""Creates coeffs in [x, y, z, w] format from quaternions
+    """
+    return q.imag.tolist() + [q.real]
+
+
+def quaternion_from_list(coeffs: np.ndarray) -> np.quaternion:
+    r"""Creates a quaternions from coeffs in [x, y, z, w] format
+    """
+    quat = np.quaternion(0, 0, 0, 0)
+    quat.real = coeffs[3]
+    quat.imag = coeffs[0:3]
+    return quat
+
+
+def quaternion_rotate_vector(quat: np.quaternion, v: np.array) -> np.array:
+    r"""Rotates a vector by a quaternion
+    Args:
+        quaternion: The quaternion to rotate by
+        v: The vector to rotate
+    Returns:
+        np.array: The rotated vector
+    """
+    vq = np.quaternion(0, 0, 0, 0)
+    vq.imag = v
+    return (quat * vq * quat.inverse()).imag
+
+
+def agent_state_target2ref(
+    ref_agent_state: List, target_agent_state: List
```

Comment:
Typing seems a bit too strict here, won't this error if fed in an np.array or a tuple which should also both be valid?

Reply:
Hi, I think np.array may not be a good idea since the lengths of `position` and `rotation` are different. If we want to specify the np.array, an alternative way is to separate a single argument `agent_state` into two, namely `position` and `rotation`. How do you like this?

Reply:
Oh, right forgot they had different lengths. Just make it a ```Union[List,Tuple]```